src/services/user.service.ts

```diff
@@ -1,49 +1,62 @@
+import { Op } from "sequelize";
+
 import { IUser } from "../models";
-import { UserRepository } from "../repositories";
+import { User } from "../models";
 
 export class UserService {
-  constructor(private readonly userRepository: UserRepository) {}
-
-  getUsers(): IUser[] {
-    return this.userRepository.getUsers();
+  async getUsers(): Promise<User[]> {
```

Comment:
I would use toJSON to sequelize data object and IUser interface

Reply:
toDomain method has been added to the User class and used to convert the User model to the IUser interface.